engine/src/main/java/com/google/android/fhir/impl/FhirEngineImpl.kt

```diff
@@ -179,6 +179,17 @@ internal class FhirEngineImpl(private val database: Database, private val contex
     }
   }
 
+  /**
+   * FHIR uses weak ETag that look something like W/"MTY4NDMyODE2OTg3NDUyNTAwMA", so we need to
+   * extract version from it. See https://hl7.org/fhir/http.html#Http-Headers.
+   */
+  private fun getVersionFromETag(eTag: String) =
+    if (eTag.startsWith("W/")) {
```

Comment:
nit : can you please check below suggestion.
```suggestion
private companion object {
    // E.G  ETag: W/"3141"
    private const val ETAG_PREFIX = "W/"
    private const val ETAG_VERSION_ID_DELIMITER = "\""
    private const val ETAG_VERSION_ID_INDEX = 1
  }
 ...
    if (eTag.startsWith(ETAG_PREFIX)) {
      eTag.split(ETAG_VERSION_ID_DELIMITER)[ETAG_VERSION_ID_INDEX]
    } else {
 ...
    
```